### ISSUE TITLE
server: secure cluster: initial screen loads, every ajax request fails with 500 internal server error

### ISSUE THREAD

**Issue by oliverkofoed** [cockroachdb/cockroach#10873]
- What did you do: Trying to start a secure cluster with only one node.
- What did you expect to see: The admin ui working
- What did you see instead: The admin ui doesn't work. it loads but every ajax request fails with 500 internal server error.

This issue is the same as was reported in #10120 

screenshot:
![screen shot 2016-11-20 at 4 54 19 pm](https://cloud.githubusercontent.com/assets/313632/20467794/8ee475c8-af43-11e6-94d0-a7053365e111.png)

Environment: Cockroach running inside docker inside a virtual box on a mac.

Docker is started with:
`docker run --detach --label dogo=13be743be94f2b66100b4e81b6096be4ae109972 --name cockroachdb --restart always --ulimit nofile=20000:20000 -p 26257:26257 -p 8080:8080 -v /data/cockroachdb:/var/opt/cockroachdb sha256:af40fcfe51bd1f7952a86518f7fbdcfa8c22ef9ca2042029780b6ab94704cb33 /opt/cockroachdb/cockroach start --logtostderr --ca-cert=/var/opt/cockroachdb/ca.cert --key=/var/opt/cockroachdb/node.key --cert=/var/opt/cockroachdb/node.cert --attrs=dc_westcoast --advertise-host=192.168.200.21 --store=/var/opt/cockroachdb`

The docker file is:
```
FROM debian:jessie

# Install packages needed for CockroachDB
RUN apt-get update && \
    DEBIAN_FRONTEND="noninteractive" && \
	apt-get install -y wget && \
	apt-get clean && \
	rm -rf /var/lib/apt/lists/*

# install CockroachDB
RUN wget https://binaries.cockroachdb.com/cockroach-beta-20161103.linux-amd64.tgz && \
	tar xfz cockroach-beta-20161103.linux-amd64.tgz && \
	rm cockroach-beta-20161103.linux-amd64.tgz && \
	mv cockroach* /opt/cockroachdb

EXPOSE 26257 

EXPOSE 8080 
```

ifconfig for the virtual machine:
[ifconfig.txt](https://github.com/cockroachdb/cockroach/files/602472/ifconfig.txt)

Certificates:
[Certs.zip](https://github.com/cockroachdb/cockroach/files/602473/Certs.zip)

Logfile:
[log.txt](https://github.com/cockroachdb/cockroach/files/602474/log.txt)

One thing i notice is that the log file says: 
> admin:      https://ea88c5283a23:8080
But i haven't specified IPV6 anywhere, so i guess it's picking it up somewhere. I would assume it would use the advertise-host option from above.

**Comment by knz** [comment]
cc @mberhault

**Comment by knz** [comment]
fyi @oliverkofoed the address `ea88c5283a23` is not an ipv6 address; it is a regular host name. I believe this would be the reverse-lookup for the IP address you have used for --host or --advertise-host.

**Comment by bdarnell** [comment]
`ea88c5283a23` looks like the fake hostnames that docker auto-generates.

**Comment by oliverkofoed** [comment]
I tried going to https://ea88c5283a23:8080 which didn't work. (I don't think it even opened the page)

**Comment by mberhault** [comment]
Could you try setting `--http-host=192.168.100.20`, this is the host the web server will listen on.
`--advertise-host` is the host told to other nodes (basically, how should other nodes reach this one). You should keep it too.

**Comment by mrtracy** [comment]
@sploiselle I believe you found a resolution for a similar issue?  I don't see it recorded in #10374 or #10120

**Comment by sploiselle** [comment]
@oliverkofoed I fixed the problem by generating certificates that included the machine's hostname. You can see an example of this [in our docs](https://www.cockroachlabs.com/docs/deploy-cockroachdb-on-digital-ocean.html#step-3-generate-your-certificates).

**Comment by oliverkofoed** [comment]
that's kinda hard to do given it's being run inside a docker container, and not a static host, no?

**Comment by sploiselle** [comment]
@oliverkofoed Not familiar enough with Docker, but @a-robinson might be.

Alex: Could you issue a node certificate including a hostname in Docker?

**Comment by tamird** [comment]
`docker run` takes a `-h, --hostname string` flag which can be used to set the container's hostname.

**Comment by a-robinson** [comment]
@tamird's suggestion is the best workaround I'm aware of for this.

We should really fix #10374 soon, though, since needing to specify the container hostname is just annoying friction that we shouldn't be requiring of people.

**Comment by EamonZhang** [comment]
@tamird 
Yes , I fix this problem by add  --host flag when start server.

**Comment by mrtracy** [comment]
This is not a UI issue, it is a server issue.  I'm not sure if it even is an issue; was this properly resolved?  It seems that #10374 already captures the only suggested improvement.

**Comment by tamird** [comment]
Indeed, this is tracked in #10374.
